qiskit/aqua/algorithms/phase_estimators/hamiltonian_pe.py

```diff
@@ -0,0 +1,165 @@
+    present algorithm, we are not actually considering time evolution. Rather, the role of time is
+    played by the scaling factor, which is chosen to best extract the eigenvalues of the
+    Hamiltonian.
+    """
+    def __init__(self,
+                 num_evaluation_qubits: int,
+                 hamiltonian: OperatorBase,
+                 evolution: Optional[Union[EvolutionBase, TempPauliEvolve]] = None,
+                 state_preparation: Optional[QuantumCircuit] = None,
+                 bound: Optional[float] = None,
+                 quantum_instance: Optional[Union[QuantumInstance, BaseBackend]] = None) -> None:
+        """Args:
+            num_evaluation_qubits: The number of qubits used in estimating the phase. The
+                                   phase will be estimated as a binary string with this many
+                                   bits.
+            hamiltonian: a Hamiltonian or Hermitian operator
+            evolution: An evolution object that generates a unitary from `hamiltonian`.
+            state_preparation: The circuit that prepares the state whose eigenphase will be
+                                 measured.  If this parameter is omitted, no preparation circuit
+                                 will be run and input state will be the all-zero state in the
+                                 computational basis.
+            bound: An upper bound on the absolute value of the eigenvalues of
+                `hamiltonian`. If omitted, and `hamiltonian` is a Pauli sum, then a bound will be
+                computed.
+            quantum_instance: The quantum instance on which the circuit will be run.
+        """
+
+        self._hamiltonian = hamiltonian
+        self._evolution = evolution
+        self._bound = bound
+
+        self._set_scale()
+        unitary = self._get_unitary()
+
+        super().__init__(num_evaluation_qubits,
+                         unitary=unitary,
+                         pe_circuit=None,
+                         num_unitary_qubits=None,
+                         state_preparation=state_preparation,
+                         quantum_instance=quantum_instance)
+
+    def _set_scale(self) -> None:
+        if self._bound is None:
+            pe_scale = phase_estimation_scale.from_pauli_sum(self._hamiltonian)
+            self._pe_scale = pe_scale
+        else:
+            self._pe_scale = PhaseEstimationScale(self._bound)
+
+    def _get_unitary(self) -> QuantumCircuit:
+        """Evolve the Hamiltonian to obtain a unitary.
+
+        Apply the scaling to the Hamiltonian that has been computed from an eigenvalue bound
+        and compute the unitary by applying the evolution object.
+        """
+        # scale so that phase does not wrap.
+        scaled_hamiltonian = self._pe_scale.scale * self._hamiltonian
+        unitary = self._evolution.convert(scaled_hamiltonian.exp_i())
+        if not isinstance(unitary, qiskit.QuantumCircuit):
```

Comment:
```suggestion
        if not isinstance(unitary, QuantumCircuit):
```

Reply:
Due to cutting and pasting, etc., I am mixing two approaches to importing symbols here. In general, I prefer to fully, or mostly, qualify symbols when they are used rather than importing the leaf symbol. With complicated code, this makes it easier to understand and modify. But, some symbols, such as `QuantumCircuit` are so ubiquitous that it seems silly to qualify them everywhere.

Reply:
I removed two of these and associated import